aesara/graph/fg.py

```diff
@@ -249,6 +262,39 @@ def add_client(self, var: Variable, new_client: Tuple[Apply, int]) -> None:
         """
         self.clients[var].append(new_client)
 
+    def replace_node(self, node, new_node, i):
+        old_var = node.inputs[i]
+        new_var = new_node.inputs[i]
+
+        self.apply_nodes.remove(node)
+        self.apply_nodes.add(new_node)
+
+        # The variable might already be removed
+        # (for instance in case of double inputs)
+        if old_var in self.variables:
+            self.variables.remove(old_var)
+        self.variables.add(new_var)
+
+        curr_clients = self.clients.pop(node.out)
+        self.setup_var(new_node.out)
+
+        for old_node, idx in curr_clients:
+            if old_node == "output":
+                # Handle graph outputs seperately
+                self.add_client(new_node.out, ("output", idx))
+                self.outputs.remove(node.out)
+                self.outputs.append(new_node.out)
+                self.variables.remove(node.out)
+                self.variables.add(new_node.out)
+                continue
+
```

Comment:
`else`?

Reply:
Ah the below part of the `for` loop is the `else` part, I added it like this since `"output"` was supposed to be a special case or something along those lines. Anyway I'll put in an else just to make it clear.